File: libs/bus-core/eventsub/eventsub.go
package eventsub

import platformentity "github.com/twirapp/twir/libs/entities/platform"

const (
	EventsubSubscribeAllSubject = "eventsub.subscribeAll"
	EventsubSubscribeSubject    = "eventsub.subscribe"
	EventsubInitChannelsSubject = "eventsub.initChannels"
	EventsubUnsubscribeSubject  = "eventsub.unsubscribe"
)

type TransportKind string

const (
	TransportWebhook   TransportKind = "webhook"
	TransportWebSocket TransportKind = "websocket"
)

type EventsubSubscribeToAllEventsRequest struct {
	ChannelID string
	Platform  platformentity.Platform
}

type EventsubSubscribeRequest struct {
	ChannelID string
	Topic     string
	Version   string
}

type EventsubBindingSnapshot struct {
	ID                string
	UserID            string
	PlatformChannelID string
}

type EventsubUnsubscribeRequest struct {
	ChannelID string
	Platform  platformentity.Platform
	Binding   *EventsubBindingSnapshot
}

File: libs/bus-core/src/eventsub/eventsub.ts
export const EventsubSubscribeAllSubject = 'eventsub.subscribeAll'
export const EventsubUnsubscribeSubject = 'eventsub.unsubscribe'

export interface EventsubSubscribeToAllEventsRequest {
	readonly ChannelID: string
	readonly Platform: string
}

export interface EventsubBindingSnapshot {
	readonly ID: string
	readonly UserID: string
	readonly PlatformChannelID: string
}

export interface EventsubUnsubscribeRequest {
	readonly ChannelID: string
	readonly Platform: string
	readonly Binding?: EventsubBindingSnapshot
}
